service/src/test/java/uk/nhs/adaptors/gp2gp/uat/EhrExtractUATTest.java

```diff
@@ -167,6 +172,47 @@ public void When_MappingValidJsonRequestBody_Expect_ValidXmlOutput(String inputJ
 
         assertThatCode(() -> validateFileContentAgainstSchema(hl7TranslatedResponse))
             .doesNotThrowAnyException();
+
+        assertThatAgentReferencesAreValid(hl7TranslatedResponse);
+    }
+
+    @SneakyThrows
+    private void assertThatAgentReferencesAreValid(String hl7) {
+        var xPathService = new XPathService();
+        var document = xPathService.parseDocumentFromXml(hl7);
+        var agentRefIdNodes = xPathService.getNodes(document, "//agentRef/id");
+        Set<String> referencedAgentIds = extractIdsFromNodeList(agentRefIdNodes, true);
+        var agentIdNodes = xPathService.getNodes(document, "//Agent/id");
+        Set<String> agentIds = extractIdsFromNodeList(agentIdNodes, false);
+
+        assertThat(referencedAgentIds).isNotEmpty();
+        assertThat(agentIds).isNotEmpty();
+        referencedAgentIds.stream()
+            .filter(id -> !"nullFlavor".equals(id))
```

Comment:
I believe a reference to "nullFlavor" is incorrect - "nullFlavor" is not a valid agent id

Reply:
These are coming from `NonConsultationResourceMapper` lines 178-182
```
        if (!isAgentPerson) {
            diagnosticReportXml
                .author(NULL_FLAVOR)
                .participant2(NULL_FLAVOR);
        }
```